presentation/src/main/java/com/knocklock/presentation/lockscreen/LockScreenStateHolder.kt

```diff
@@ -1,83 +1,106 @@
+    )
     val notificationList = _notificationList.asStateFlow()
 
-    private val useCaseEntryPoint =
-        EntryPointAccessors.fromApplication(
-            context,
-            UseCaseEntryPoint::class.java
-        )
+    private val packageManager by lazy { context.packageManager }
 
-    val getNotificationUseCase = useCaseEntryPoint.getNotificationUseCase()
-    val deleteAllNotificationUseCase = useCaseEntryPoint.deleteAllNotificationUseCase()
+    fun updateNotificationArray(notificationArray: Array<StatusBarNotification>) {
+        val notificationUiState = NotificationUiState.Success(
+            notificationList = notificationArray.map { statusBarNotification ->
+                with(statusBarNotification.notification.extras) {
+                    var appTitle = ""
+                    val subText: String = convertString(getCharSequence("android.subText"))
+                    val title: String = convertString(getCharSequence("android.title"))
+                    val content: String = convertString(getCharSequence("android.text"))
+                    val packageName = statusBarNotification.packageName
+                    val date = Date(statusBarNotification.postTime)
+                    val stringPostTime = try {
+                        SimpleDateFormat("a HH:mm", Locale.KOREA).format(date)
+                    } catch (e: Exception) {
+                        ""
+                    }
+                    val applicationInfo: ApplicationInfo?
 
-    init {
-        deleteAllNotification()
-        getNotification()
-    }
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+                        applicationInfo = try {
+                            packageManager.getApplicationInfo(
+                                packageName,
+                                PackageManager.ApplicationInfoFlags.of(PackageManager.GET_META_DATA.toLong())
+                            )
+                        } catch (e: PackageManager.NameNotFoundException) {
+                            null
+                        }
+                        if (applicationInfo == null) {
+                            println("package name : unknown")
+                        } else {
+                            appTitle = packageManager.getApplicationLabel(applicationInfo).toString()
+                        }
+                    } else {
+                        applicationInfo = try {
+                            packageManager.getApplicationInfo(packageName, 0)
+                        } catch (e: PackageManager.NameNotFoundException) {
+                            null
+                        }
+                        if (applicationInfo == null) {
+                            println("package name : unknown")
+                        } else {
+                            appTitle = packageManager.getApplicationLabel(applicationInfo).toString()
+                        }
+                    }
```

Comment:
```suggestion
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
                        applicationInfo = runCatching {
                            packageManager.getApplicationInfo(
                                packageName,
                                PackageManager.ApplicationInfoFlags.of(PackageManager.GET_META_DATA.toLong())
                            )
                        }.onSuccess {
                            appTitle =
                                packageManager.getApplicationLabel(it).toString()
                        }.getOrNull()
                    } else {
                        applicationInfo = runCatching {
                            packageManager.getApplicationInfo(packageName, 0)
                        }.onSuccess {
                            appTitle =
                                packageManager.getApplicationLabel(it).toString()
                        }.getOrNull()
                    }
```
application이 성공일 때도 null이 들어올 수 있다면  기본 코드로 유지해도 되겠지만, 그게 아니라면 분기를 줄이고 가독성 면에서 runCatching을 추천드립니다!

Reply:
@KimHance 맛있네요 반영해볼께요

Reply:
@KimHance [eded568](https://github.com/Knock-Lock/Knock-Lock/pull/75/commits/eded56836f8a493fc42847e89e954180ea16edab) 반영했습니다